pkg/ns/ns.go

```diff
@@ -179,7 +191,28 @@ func (ns *netNS) Close() error {
 	return nil
 }
 
+func (ns *netNS) validate(action string) error {
+	switch {
+	case ns.file == nil:
+		return fmt.Errorf(
+			"%q failed: no filedescriptor associated",
+			action,
+		)
+	case ns.mounted == false:
```

Comment:
We don't care if some paths aren't mounted; we only care that ones created through NewNS() are mounted.  For example, a CNI plugin that is passed a namespace path needs to close the file handle with Close(), and if mounted:true then pkg/ns will also try to unmount it and remove the mountpoint when Close() is called.  We only want that to happen for things that pkg/ns creates with NewNS(), not for any random namespace passed into the process.